oida/commands/componentize.py

```diff
@@ -185,3 +195,93 @@ class {class_name}(AppConfig):
+            new_module_name_list = self.new_module.split(".")
+            # The full module name is in the form: component.app.subdir.subdir.module
+            # while the module we are changing is of the form: project.component.app
+            # We, therefore, need to check if the last two (string) components of the name of the
+            # new module are equal to the first two (string) components of the name of the module
+            # currently being processed. If we do not do this test - all tasks in the project
+            # (e.g. tienda) will be prepended with the component name, which is wrong.
+            if (
+                len(current_module_name_list) >= 2
+                and len(new_module_name_list) >= 2
+                and current_module_name_list[:2] == new_module_name_list[-2:]
+            ):
+                return tree.visit(self)
+        return tree
+
+    def update_decorator(
+        self, decorator: cst.Decorator, task_name: str
+    ) -> cst.Decorator:
+        # Test if the name of the task is not explicitly set
+        if not m.matches(
+            decorator,
+            m.Decorator(
+                decorator=m.Call(
+                    args=[m.ZeroOrMore(), m.Arg(keyword=m.Name("name")), m.ZeroOrMore()]
+                ),
+            ),
+        ):
+            call = decorator.decorator
+            if isinstance(call, cst.Call):
+                arguments = call.args
+                arguments = (  # type: ignore
+                    cst.Arg(
+                        keyword=cst.Name("name"),
+                        value=cst.SimpleString(value=task_name),
+                    ),
+                ) + arguments
+                new_call = call.with_changes(args=arguments)
+                new_decorator = decorator.with_changes(decorator=new_call)
+                return new_decorator
+        return decorator
+
+    def leave_FunctionDef(
+        self, original_node: cst.FunctionDef, updated_node: cst.FunctionDef
+    ) -> BaseStatement | FlattenSentinel[BaseStatement] | RemovalSentinel:
+        celery_task_decorator = m.Decorator(
+            m.Call(m.Attribute(value=m.Name("app"), attr=m.Name("task")))
+        )
+        # The implicit name of the task is the path to the old module concatenated with the function name.
+        task_name = f'"{self.old_module}.{original_node.name.value}"'
+        decorators = [
+            self.update_decorator(decorator=decorator, task_name=task_name)
+            if m.matches(decorator, celery_task_decorator)
+            else decorator
+            for decorator in updated_node.decorators
+        ]
+        return updated_node.with_changes(decorators=decorators)
+
+
+def update_celery_task_names(root_module: Path, old_path: Path, new_path: Path) -> None:
+    old_module = get_module(old_path)
+    new_module = get_module(new_path)
+
+    files = [str(path) for path in root_module.rglob("*.py")]
```

Comment:
```suggestion
    files = [str(path) for path in new_path.rglob("*.py")]
```

As noted above, no need for running this across all files in the repo, we just need to look at the module we've moved.